# LOAN DATASET ANALYSIS -practise
**-MOHAMED MAHIR**

**here iam going to analyse the loan dataset on how people are getting their loan aproved** 

In [65]:
#lets import the libraries
import pandas as pd
import numpy as np 


In [66]:
#lets read the csv file
loan_df=pd.read_csv("./datasets/loan.csv")

In [67]:
#taking a look at the dataset with first 5 rows
loan_df.head()

Loan_ID Gender Married Dependents     Education Self_Employed  \
0  LP001002   Male      No          0      Graduate            No   
1  LP001003   Male     Yes          1      Graduate            No   
2  LP001005   Male     Yes          0      Graduate           Yes   
3  LP001006   Male     Yes          0  Not Graduate            No   
4  LP001008   Male      No          0      Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5849                0.0         NaN             360.0   
1             4583             1508.0       128.0             360.0   
2             3000                0.0        66.0             360.0   
3             2583             2358.0       120.0             360.0   
4             6000                0.0       141.0             360.0   

   Credit_History Property_Area Loan_Status  
0             1.0         Urban           Y  
1             1.0         Rural           N  
2             1.0         Urban           Y  
3             1.0         Urban           Y  
4             1.0         Urban           Y

In [68]:
#checking the shape
loan_df.shape

(614, 13)

In [69]:
#some info about the dataset
loan_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


**there seems to be some missing values in the data**

In [70]:
#checking the null values
loan_df.isna().sum()

Loan_ID               0
Gender               13
Married               3
Dependents           15
Education             0
Self_Employed        32
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount           22
Loan_Amount_Term     14
Credit_History       50
Property_Area         0
Loan_Status           0
dtype: int64

**since there are only less count of null values in gender and married we can droo the null rows**

In [71]:
#droping the rows with null values in gender column
loan_df.dropna(subset=["Gender","Married"],inplace=True)

**lets consider the null values in dependants as 0 considering they have no dependents**

In [72]:
loan_df.Dependents.fillna("0",inplace=True)

**the null values in selfemployed column can be filled with "no data"**

In [73]:
loan_df.Self_Employed.fillna("no_data",inplace=True)

**we can fill the null values in loan amount term with frequent values**

In [74]:
loan_df.Loan_Amount_Term.mode()

0    360.0
dtype: float64

In [75]:
#since 360 is most frequently occured we can fill the null values with that
loan_df.Loan_Amount_Term.fillna(360,inplace=True)

**we have dealt with the null values ,now lets see some info on the numerical and non numerical columns**

In [76]:
#stastical info on the numerical columns
round(loan_df.describe())

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count            598.0              598.0       577.0             598.0   
mean            5292.0             1631.0       145.0             342.0   
std             5807.0             2953.0        83.0              64.0   
min              150.0                0.0         9.0              12.0   
25%             2878.0                0.0       100.0             360.0   
50%             3806.0             1212.0       127.0             360.0   
75%             5746.0             2324.0       167.0             360.0   
max            81000.0            41667.0       650.0             480.0   

       Credit_History  
count           549.0  
mean              1.0  
std               0.0  
min               0.0  
25%               1.0  
50%               1.0  
75%               1.0  
max               1.0

as you can see,

    >average loanamount is 145
    >75% amount of people have credit history


In [77]:
#statiscal info on the other columns
loan_df.describe(include="object")

Loan_ID Gender Married Dependents Education Self_Employed  \
count        598    598     598        598       598           598   
unique       598      2       2          4         2             3   
top     LP001002   Male     Yes          0  Graduate            No   
freq           1    487     388        350       465           488   

       Property_Area Loan_Status  
count            598         598  
unique             3           2  
top        Semiurban           Y  
freq             225         411

From this we can see that,

    >Male people have applied for loan more times than female
    >Most of the people are graduate
    >most of the people are not self employed
    >most people are applying for long term loans

we have got some **insights** on the type of people who are applying for the loan

now lets do some analysis for people getting **loan approved**

In [78]:
#taking only the rows for approved loans
loan_y_df=loan_df[loan_df.Loan_Status=="Y"]
loan_y_df.head()

Loan_ID Gender Married Dependents     Education Self_Employed  \
0  LP001002   Male      No          0      Graduate            No   
2  LP001005   Male     Yes          0      Graduate           Yes   
3  LP001006   Male     Yes          0  Not Graduate            No   
4  LP001008   Male      No          0      Graduate            No   
5  LP001011   Male     Yes          2      Graduate           Yes   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5849                0.0         NaN             360.0   
2             3000                0.0        66.0             360.0   
3             2583             2358.0       120.0             360.0   
4             6000                0.0       141.0             360.0   
5             5417             4196.0       267.0             360.0   

   Credit_History Property_Area Loan_Status  
0             1.0         Urban           Y  
2             1.0         Urban           Y  
3             1.0         Urban           Y  
4             1.0         Urban           Y  
5             1.0         Urban           Y

In [79]:
#grouping for male and female
loan_y_df.groupby("Gender")["Loan_Status"].count()

Gender
Female     74
Male      337
Name: Loan_Status, dtype: int64

from the ^above results,

    >male people are getting aproved for loan more than female but this cannot be completely true as males are applying for loans more than females in this datset

In [80]:
#grouping for education 
loan_y_df["Education"].value_counts()

Graduate        329
Not Graduate     82
Name: Education, dtype: int64

from the ^above resluts,
    
    >people who are graduated are getting approved way more than not educated  

In [81]:
#having graduate people grouping for self employed
loan_y_df["Self_Employed"][loan_y_df.Education=="Graduate"].value_counts()

No         269
Yes         42
no_data     18
Name: Self_Employed, dtype: int64

    >most people are not self employed therfore this does not play a major factor in getting loan approved

In [82]:
#having gradute people grouping for their martial status
loan_y_df["Married"][loan_y_df.Education=="Graduate"].value_counts()

Yes    225
No     104
Name: Married, dtype: int64

    >most people are married if they are married and graduated

In [83]:
loan_y_df["Married"].value_counts()

Yes    279
No     132
Name: Married, dtype: int64

    >regardless of their education married people are getting loan approved more than unmarried people

In [84]:
loan_y_df.Credit_History.value_counts()

1.0    368
0.0      7
Name: Credit_History, dtype: int64

    >if they do not have a  credit history thier chances of getting loan approved is very low

In [85]:
#creating pivot table for loan amount term
pd.crosstab(loan_df["Loan_Amount_Term"],loan_df["Loan_Status"])

Loan_Status         N    Y
Loan_Amount_Term          
12.0                0    1
36.0                2    0
60.0                0    2
84.0                1    3
120.0               0    3
180.0              15   28
240.0               1    2
300.0               5    7
360.0             154  360
480.0               9    5

from ^ above ,

    >we can see it does not play a major role in getting loan approved but most of people applied for long term loans 

In [86]:
#pivot table for dependents
pd.crosstab(loan_df.Dependents,loan_df.Loan_Status)

Loan_Status    N    Y
Dependents           
0            110  240
1             36   65
2             23   76
3+            18   30

if they have less number of **dependants** or **no dependants** their chance of getting loan approved is more

In [99]:
#pivot table for property area
print("\n\nNo of people in each area based loan status")
print(pd.crosstab(loan_df.Property_Area,loan_df.Loan_Status))
print("\n\nTotal no of people in each area")
print(loan_df.Property_Area.value_counts())



No of people in each area based loan status
Loan_Status     N    Y
Property_Area         
Rural          67  108
Semiurban      52  173
Urban          68  130


Total no of people in each area
Semiurban    225
Urban        198
Rural        175
Name: Property_Area, dtype: int64


**semiurban**  area people are getting loans approved comparitively more and they are the people who  are applying for loans more but **urban and rural** peolple also getting loans approved even though it is less comparable to semiurban

In [106]:
loan_df.ApplicantIncome.describe()

count      598.000000
mean      5292.252508
std       5807.265364
min        150.000000
25%       2877.500000
50%       3806.000000
75%       5746.000000
max      81000.000000
Name: ApplicantIncome, dtype: float64

In [113]:
#applicant income less thn 3k
a=loan_df[loan_df.ApplicantIncome.between(0,3000)]
#applicant income beetween 3k to 6k
b=loan_df[loan_df.ApplicantIncome.between(3000,6000)]
#apllicant income more than 6k
c=loan_df[loan_df.ApplicantIncome>6000]

In [120]:
print(a.Loan_Status.value_counts())
print("total entries between 0 to 3k income : ",len(a))

Y    121
N     52
Name: Loan_Status, dtype: int64
total entries between 0 to 3k income :  173


In [121]:
print(b.Loan_Status.value_counts())
print("total entries between 3k to 6k : ",len(b))

Y    207
N     90
Name: Loan_Status, dtype: int64
total entries between 3k to 6k :  297


In [122]:
print(c.Loan_Status.value_counts())
print("total entries more than 6k : ",len(c))

Y    85
N    46
Name: Loan_Status, dtype: int64
total entries more than 6k :  131


from the above analysis there seems to be no difference in loans getting approved based on thier annual income